# Задачи по Эконометрике: интерпретация коэффициентов 


# Cost Equation

## задача 1

#### рассмотрим регрессию log(cost) на log(q) на датсете electricity

#### Спецификация:
$$
log(cost)= \beta_0 + \beta_1*log(q) + u
$$

In [5]:
import numpy as np # linear algebra 
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
# stats models: regression fitting via formulas
import statsmodels.formula.api as smf
# stats models: regression fitting via matrices of regression design
import statsmodels.api as sm

In [11]:

df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/Electricity.csv')
df


cost        q        pl      sl      pk      sk      pf      sf
0     0.2130      8.0   6869.47  0.3291  64.945  0.4197  18.000  0.2512
1     3.0427    869.0   8372.96  0.1030  68.227  0.2913  21.067  0.6057
2     9.4059   1412.0   7960.90  0.0891  40.692  0.1567  41.530  0.7542
3     0.7606     65.0   8971.89  0.2802  41.243  0.1282  28.539  0.5916
4     2.2587    295.0   8218.40  0.1772  71.940  0.1623  39.200  0.6606
..       ...      ...       ...     ...     ...     ...     ...     ...
153   6.8293    946.6  10642.16  0.0883  43.600  0.1914  51.463  0.7203
154   3.7605    377.0   7432.24  0.2117  74.120  0.2274  33.436  0.5609
155   3.9822    391.0   5826.04  0.1926  78.288  0.0924  44.633  0.7151
156  30.1880   5317.0   9586.63  0.0845  78.008  0.2009  41.840  0.7147
157  67.8562  15220.0   6986.09  0.1451  74.025  0.2596  25.662  0.5945

[158 rows x 8 columns]

In [12]:
### оценивание через матрицы регрессивного дизайна Y, X

In [13]:
# Зададим матрицы регрессионного дизайна
y = np.log(df['cost']) # зависимая переменная
# логаримируем данные
X = pd.DataFrame(data={'l_q': np.log(df['q'])} )
X = sm.add_constant(X)
#X = sm.add_constant(df[['totwrk', 'male']]) # регрессоры с добавленным столбом единиц
X.head(n=7)

const       l_q
0    1.0  2.079442
1    1.0  6.767343
2    1.0  7.252762
3    1.0  4.174387
4    1.0  5.686975
5    1.0  5.209486
6    1.0  3.912023

In [14]:
cost_eq2 = sm.OLS(y, X).fit()
# Коэфициенты модели с округлением
cost_eq2.params.round(3)

const   -3.841
l_q      0.836
dtype: float64

### При увеличении выпуска электроэнергии на 1%, издержки за год увеличиваются на 0.836%


## модель 2

## спецификация: 
$$
log(cost)= \beta_0 + \beta_1 * log(q) + \beta_2 * log(pl) + \beta_3 * log(pf) + \beta_4 * log(pk) + u
$$

In [15]:
# Зададим матрицы регрессионного дизайна
y = np.log(df['cost']) 
# логаримируем данные
X = pd.DataFrame(data={'l_q': np.log(df['q']), 'l_pl': np.log(df['pl']), 'l_pf': np.log(df['pf']), 'l_pk': np.log(df['pk'])} )
X = sm.add_constant(X)
#X = sm.add_constant(df[['totwrk', 'male']]) 
X.head(n=7)

const       l_q      l_pl      l_pf      l_pk
0    1.0  2.079442  8.834842  2.890372  4.173541
1    1.0  6.767343  9.032763  3.047708  4.222840
2    1.0  7.252762  8.982297  3.726416  3.706032
3    1.0  4.174387  9.101852  3.351272  3.719481
4    1.0  5.686975  9.014131  3.668677  4.275832
5    1.0  5.209486  8.529811  3.569814  4.309859
6    1.0  3.912023  9.127420  3.467921  4.505018

In [16]:
cost_eq2 = sm.OLS(y, X).fit()
# Коэфициенты модели с округлением
cost_eq2.params.round(3)

const   -7.472
l_q      0.838
l_pl     0.044
l_pf     0.713
l_pk     0.188
dtype: float64

##### При увеличении выпуска электроэнергии на 1%, издержки за год увеличиваются на 0.838%